sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/planner/BeamRuleSets.java

```diff
@@ -39,22 +60,91 @@
+      RuleSets.ofList(
+          // translate to beam logical rel nodes
+          BeamAggregationRule.INSTANCE,
+          BeamEnumerableConverterRule.INSTANCE,
+          BeamFilterRule.INSTANCE,
+          BeamIntersectRule.INSTANCE,
+          BeamIOSinkRule.INSTANCE,
+          BeamJoinRule.INSTANCE,
+          BeamMinusRule.INSTANCE,
+          BeamProjectRule.INSTANCE,
+          BeamSortRule.INSTANCE,
+          BeamUnionRule.INSTANCE,
+          BeamUncollectRule.INSTANCE,
+          BeamUnnestRule.INSTANCE,
+          BeamValuesRule.INSTANCE,
+
+          // push a filter into a join
+          FilterJoinRule.FILTER_ON_JOIN,
+          // push filter into the children of a join
+          FilterJoinRule.JOIN,
+          // push filter through an aggregation
+          FilterAggregateTransposeRule.INSTANCE,
+          // push filter through set operation
+          FilterSetOpTransposeRule.INSTANCE,
+          // push project through set operation
+          ProjectSetOpTransposeRule.INSTANCE,
+
+          // push a projection past a filter or vice versa
+          ProjectFilterTransposeRule.INSTANCE,
+          FilterProjectTransposeRule.INSTANCE,
+
+          // push a projection to the children of a join
+          // push all expressions to handle the time indicator correctly
+          new ProjectJoinTransposeRule(
+              PushProjector.ExprCondition.FALSE, RelFactories.LOGICAL_BUILDER),
+          // merge projections
+          ProjectMergeRule.INSTANCE,
+          // reorder sort and projection
+          SortProjectTransposeRule.INSTANCE,
+          ProjectSortTransposeRule.INSTANCE,
+
+          // join rules
+          JoinPushExpressionsRule.INSTANCE,
+
+          // remove union with only a single child
+          UnionEliminatorRule.INSTANCE,
+          // convert non-all union into all-union + distinct
+          UnionToDistinctRule.INSTANCE,
+
+          // remove aggregation if it does not aggregate and input is already distinct
+          AggregateRemoveRule.INSTANCE,
+          // push aggregate through join
+          AggregateJoinTransposeRule.EXTENDED,
+          // aggregate union rule
+          AggregateUnionAggregateRule.INSTANCE,
+
+          // remove unnecessary sort rule
+          SortRemoveRule.INSTANCE,
+
+          // prune empty results rules
+          PruneEmptyRules.AGGREGATE_INSTANCE,
+          PruneEmptyRules.FILTER_INSTANCE,
+          PruneEmptyRules.JOIN_LEFT_INSTANCE,
+          PruneEmptyRules.JOIN_RIGHT_INSTANCE,
+          PruneEmptyRules.PROJECT_INSTANCE,
+          PruneEmptyRules.SORT_INSTANCE,
+          PruneEmptyRules.UNION_INSTANCE);
 
   public static RuleSet[] getRuleSets() {
     return new RuleSet[] {
       RuleSets.ofList(
-          BeamProjectRule.INSTANCE,
-          BeamFilterRule.INSTANCE,
           BeamAggregationRule.INSTANCE,
-          BeamSortRule.INSTANCE,
-          BeamValuesRule.INSTANCE,
+          BeamEnumerableConverterRule.INSTANCE,
```

Comment:
Probably this method can just return `List<RuleSet>` and we can have separate rule sets, just for organization. Because actually the way the `RuleSet[]` is passed in `BeamQueryPlanner` is sort of abusing varargs.

I don't really know what separating it into multiple rulesets does except for being able to pass them around separately easily.